lib/Sema/SemaExpr.cpp

```diff
@@ -7401,7 +7410,19 @@ checkPointerTypesForAssignment(Sema &S, QualType LHSType, QualType RHSType) {
   // between checked and unchecked pointers or between different kinds of checked
   // pointers. Only implicit conversions between unchecked pointers or from
   // unchecked to checked pointers are allowed.
-  if (lhkind != rhkind && rhkind != CheckedPointerKind::Unchecked)
+
+  // Checked C - add rules for implicit casts between checked pointer types
+  // implicit casts between checked pointer type are not allowed
```

Comment:
This comment is confusing.     Please describe what casts are allowed now and don't worry about describing what the code used to do.